gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BigQuery.java

```diff
@@ -25,5 +34,602 @@
  */
 public interface BigQuery extends Service<BigQueryOptions> {
 
-  // TODO(mziccard) add missing methods
+  enum DatasetField {
```

Comment:
I wasn't bothered by it because I also think that name level comments are redundant. I didn't mind much about the class level as their usage is always part of a specific `Option` and they have a nice javadoc for the `fields` method however it would be nice to have a class level javadoc for consistency and for lulling checkstyle and a like about it.